Prox/Prox/Widgets/InAppNotificationToast.swift

```diff
@@ -0,0 +1,104 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import UIKit
+
+protocol InAppNotificationToastDelegate: class {
+    func inAppNotificationToastProvider(_ toast: InAppNotificationToastProvider, userDidRespondToNotificationForEventWithId eventId: String, atPlaceWithId placeId: String)
+    func inAppNotificationToastProviderDidDismiss(_ toast: InAppNotificationToastProvider)
+}
+
+class InAppNotificationToastProvider: NSObject {
+
+    fileprivate let placeId: String
+    fileprivate let eventId: String
+
+    fileprivate lazy var notificationView: UIView = {
+        let view = UIView()
+        view.backgroundColor = Colors.notificationBackground
+        view.isUserInteractionEnabled = true
+        view.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(userDidRespondToToast(recognizer:))))
+        return view
+    }()
+
+    fileprivate lazy var notificationLabel: UILabel = {
+        let label = UILabel()
+        label.numberOfLines = 0
+        label.lineBreakMode = .byWordWrapping
+        label.textColor = Colors.notificationText
+        label.font = Fonts.notificationText
+        label.backgroundColor = .clear
+        label.textAlignment = .center
+        return label
+    }()
+
+    fileprivate var presentationTimer: Timer?
+
+    weak var delegate: InAppNotificationToastDelegate?
+
+    fileprivate var viewBottomAnchor: NSLayoutConstraint!
```

Comment:
Bottom anchor isn't used for anything. Also a bit concerning to have these as explicitly unwrapped just in case!